### PR TITLE
Detect long-running requests from parsed request info

### PR DESCRIPTION
**Pull request by liggitt**
Follow up to https://github.com/kubernetes/kubernetes/pull/36064

Uses parsed request info to more tightly match verbs and subresources

Removes regex-based long-running request path matching (which is easily fooled)

```release-note
The --long-running-request-regexp flag to kube-apiserver is deprecated and will be removed in a future release. Long-running requests are now detected based on specific verbs (watch, proxy) or subresources (proxy, portforward, log, exec, attach).
```

### CODE REVIEW

**Comment by liggitt** [comment]
@gmarek PTAL

**Comment by liggitt** [comment]
cc @deads2k @sttts

File: cmd/kube-apiserver/app/server.go [hunk 1]
```diff
@@ -277,6 +279,10 @@ func Run(s *options.ServerRunOptions) error {
 	genericConfig.EnableOpenAPISupport = true
 	genericConfig.EnableMetrics = true
 	genericConfig.OpenAPIConfig.SecurityDefinitions = securityDefinitions
+	genericConfig.LongRunningFunc = filters.BasicLongRunningRequestCheck(
```
Comment (deads2k) [inline cmd/kube-apiserver/app/server.go]: Do we care about thsi being configurable at all any more? Could we just bake it into genericapiserver and anyone who wants something different can take control of the handler chain?

Comment (liggitt) [inline cmd/kube-apiserver/app/server.go]: proxy/attach/exec/portforward as subresources are only likely to make sense to the apiserver... baking those into genericapiserver didn't seem useful

**Comment by sttts** [comment]
Have only the same question as @deads2k, tending to leave the customizability as it is here. lgtm

**Comment by gmarek** [comment]
@liggitt please add release note about deprecating the flag. LGTM otherwise.

**Comment by liggitt** [comment]
set up a basic long running func for genericapiserver that knows about the `watch` verb

**Pull request merged**